src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -1467,7 +1467,7 @@ else if (!referencePath.isDirectory())
                                     String absoluteReference = objectsPath.getAbsolutePath().replace('\\', '/');
                                     listener.getLogger().println("Using reference repository: " + reference);
                                     // git implementations on windows also use
-                                    try (PrintWriter w = new PrintWriter(alternates, "UTF-8")) {
+                                    try (PrintWriter w = new PrintWriter(alternates, StandardCharsets.UTF_8)) {
```

Comment:
Using newer Java Platform functionality that avoids the messiness around `UnsupportedEncodingException`.